src/persistence/fs-persistence.ts

```diff
@@ -1,48 +1,75 @@
 import * as fs from "fs";
 import * as ini from "ini";
+import * as yml from "js-yaml";
 import * as mkdirp from "mkdirp";
+import * as path from "path";
 import { IPersistence } from "./persistence";
 
 const UNMOCK_DIR = ".unmock";
 const TOKEN_FILE = ".token";
 const CONFIG_FILE = "credentials";
-const TOKEN_PATH = `${UNMOCK_DIR}/${TOKEN_FILE}`;
-const CONFIG_PATH = `${UNMOCK_DIR}/${CONFIG_FILE}`;
+const TOKEN_PATH = path.join(UNMOCK_DIR, TOKEN_FILE);
+const CONFIG_PATH = path.join(UNMOCK_DIR, CONFIG_FILE);
+const SAVE_PATH = path.join(UNMOCK_DIR, "save");
+const RESPONSE_FILE = "response.json";
+const METADATA_FILE = "metadata.unmock.yml";
+const DATA_KEY = "body";
+const HEADER_KEY = "headers";
 
 export default class FSPersistence implements IPersistence {
   private token: string | undefined;
+
+  constructor(private savePath = SAVE_PATH) {
+  }
+
+  public saveMetadata(hash: string, data: {[key: string]: string}) {
+    const target = this.outdir(hash, METADATA_FILE);
+    // First attempt to load existing data
+    const existingData = fs.existsSync(target) ? yml.safeLoad(fs.readFileSync(target, "utf-8")) : {};
+    // Now add the new data as needed
+    for (const key of Object.keys(data)) {
+      existingData[key] = data[key];
```

Comment:
Is this somehow different than using the spread syntax like `existingData = { ...existingData, ...data }`?

Reply:
Nope! I forgot about that, thanks! :)

Reply:
Edit: Different because `existingData` is a `const`, so can't reassign to it, but otherwise it's identical.

Reply:
Perhaps it could be changed to `let? :) `Object.assign` would also be quite idiomatic js.

Reply:
`const newData = {...existingData, ...data};` :wink: